src/cart/src/cart/crt_ctl.c

```diff
@@ -154,12 +132,9 @@ crt_hdlr_ctl_get_uri_cache(crt_rpc_t *rpc_req)
 	if (rc != 0)
 		D_GOTO(out, rc);
 
-	rc = d_hash_table_traverse(&grp_priv->gp_uri_lookup_cache,
-				   crt_ctl_get_uri_cache_size_cb, &nuri);
-	if (rc != 0)
-		D_GOTO(out, 0);
-
-	D_ALLOC_ARRAY(uri_cache.grp_cache, nuri);
+	/* calculate max possible count of grp_cache items */
+	uri_cache.max_count = grp_priv->gp_size * CRT_SRV_CONTEXT_NUM;
```

Comment:
Is gp_size the correct bound on the size of the hash table, or will it be all end-points?  Unless server groups are talking to other server groups then it's probably the same thing.

Reply:
This is primary group which contain ALL ranks. Hash table use rank as key. So, Max count of different keys cannot be more than grp_priv->gp_size. So, I think this is correct maximum assumption.